File: server/Domain/Enums/RelationShip/MembershipStatus.cs
namespace Domain.Enums.RelationShip;

public enum MembershipStatus
{
    Pending,   // User requested to join with a code, waiting for approval
    Active,    // Approved member
    Invited,   // Invited directly by an admin, hasn't accepted yet
    Suspended  // Member was suspended by an admin
}

File: frontend/src/types/membership-status.ts
export type MembershipStatus = "Pending" | "Active" | "Invited" | "Suspended";

export const MEMBERSHIP_STATUS_LABELS: Record<MembershipStatus, string> = {
  Pending: "Pending",
  Active: "Active",
  Invited: "Invited",
  Suspended: "Suspended",
};

export function getMembershipStatusLabel(status: MembershipStatus) {
  return MEMBERSHIP_STATUS_LABELS[status] ?? status;
}
